### DISCUSSION TITLE
Petraptor Companion

### DISCUSSION THREAD

**Discussion by VAHLOURRE** [FakeFishGames/Barotrauma#12110] [Ideas and suggestions]
The Petraptor, a waste of Dementonite Clusters and time, they require time and attention, if without they will starve or get lost due to a breach in the sub. Petraptors are dumb and stupid they are very useless, annoyingly noisy and is a waste of precious materials such as saline and dementonite, all they do is poop and eat while being extremely horrible at defending and attacking; in addition they have weak stats and die easily, offering nothing but liability and frustration, especially in single-player campaign.

Petraptors  must become **TRAINABLE** like dogs and have growth milestones to provide impact in the game and crew especially in **late-game and PvP**; I demand, that the Petraptor must have commands and have talent skill trees for learning talents while equipped with equipment slots for raptor-gear or equipment in order to provide the Petraptor with formidable power and ability to become a _late game exploration explorer-hard-carry_. 

In order for the Petraptor to be revolutionized the following must be implemented:

•Basic Features
-The owner must get a new hotbar available on the interface that lists commands for Petraptor to follow.

-Commands should have hierarchy, for example: protect owner, this commands must override other existing pending commands.

-The Petraptor should have radio equipment communication slot available to fill in in order to transmit commands over long distances, though some commands should be left unusable over great distances, some commands should have range limits.

-The Petraptor should have talents in order to get new skills and be able to learn to heed more commands and expand A.I.

-The Petraptor should be teleported beside the owner whenever docking and saving in an outpost and it must trigger a follow the owner command automatically, to avoid losing the Petraptor.

-The Petraptor should have collar slot and collars that can be equipped similar to head equipment slot of human entities, that serve as I.D to enter hulls with manual pet sensor modification.

-The Petraptor should have growth stages, each stage require materials, and each stage should give it more abilities, natural armor, stats, etc. To balance this hunger should increase as well.

-Stations must sell monster meat that a player can use to feed Petraptor and unique food items can give unique buffs and debuffs to Petraptor.

-Petraptors should have health interfaces, however it can only be accessed by an owner, medic or an assistant, to further balance the Petraptor medic and owner are given a -70% efficiency on using medical items on Petraptor, unless it is a assistant giving aid to the Petraptor.

-The Petraptor should have weapons slots to be filled in with weapons that can help them capture or hunt prey and fight bigger monsters, with each different types of weapons providing different uses such as: piercing, grappling, wall climbing, blunt force trauma, stun, bleeding and decapitation.
 
-The Petraptor should have artificial armor equipment in order to better protect them in combat against a swarm of monsters and bullets as well as grant them **extra speed and slowness** in land and in water similar to the **Slipsuit**. 

-The Petraptor must have a talent than will enable it to crawl between small spaces or just dash through em to provide mobility in land.

-The Petraptor must have a talent to mine ores and commands to make them breach walls like a boarding axe.

-The Petraptor should be able to be genetically modified.

-The Petraptor must never get lost, refer to the outpost teleportation mechanism above.

-The Petraptor could have an equippable brain chip that can be connected to handheld sonars in order to improve efficiency of the device.

•Combat Features

-The Petraptor should have a talent on grappling it should have two different types that will be used for bigger enemies and smaller enemies. Grappling should neutralize humans effectively and it should only happen when Petraptor sprints and jumps them like a xenomorph, it must immobilize humans and kill them my biting off their hands though effectiveness can be determined by talents, such as a hit and run grapple where the Petraptor just bites a player stunning them and drags them around as it retreats in safety or to the owner. This should also apply on smaller creatures. However, against giant creatures Petraptor should instead latch on to them similar to how mudraptors destroy the exterior of submarines. 

-The Petraptor should avoid getting damaged unless necessary and resort to hit and run maneuver that utilizes walls and environment.

-The Petraptor should have a stalk talent that enables it to walk without sounds, therefore it will be invisible to sonar.

-The Petraptor should have a talent that makes it realize it can use walls to jump on and propel themselves to deal more damage.

-**The Petraptor must win solo against a thresher swarm breach** and must utilize walls while using hit and run manuever.

•Advanced A.I Features

-Talents should improve Petraptor A.I helping it becoming human A.I.

-When out of the submarines, owner can command Petraptor to return to submarine, but due to it unable to open the airlock it will just wait and follow the submarine as it moves and latch on to the submarine.

-The Petraptor must locate the submarine without fail, similar to how wild mudraptors locate a submarine. 

-Petraptor should have a patrol sub commands, with this commands they can efficiently guard the submarine and use echolocation to successfully locate breaches and intruders, they will breach walls if necessary and protect the crew repairing the sub. **WHILE EXECUTING THIS COMMAND THEY ARE UNABLE TO GET OUT OF SUBMARINE AT ALL COST.** 

That Petraptor should be a formidable ally for the game and crew, like **Rex** the **Tamed Armored Mudraptor** from the civilian transportation event, big and scary, strong and deadly, the Petraptor will be your friend and ally amongst the denizens of Europa. A biological weapon able to tear monsters and humans apart in one single command or provocation, a force to be reckoned with.

**Comment by MrWomanStealer** [comment]
too nerdy/ur asking for too much tbh

AI changes would be a great start tho.

**Comment by G-M-twostay** [comment]
make pets such as defense bots be able to climb ladders.

**Comment by Hazmat616** [comment]
Let us ride them like scooters
